docs/deploying-cf-locally.md

```diff
@@ -0,0 +1,348 @@
+
+> 💡 Note: This environment is not production-grade.
+
+## Install VirtualBox and prerequisites
+ 
+Ensure that VirtualBox and its extension pack are installed and configured
+properly.
+
+```bash
+sudo dnf install virtualbox
+VBoxManage extpack install --replace Oracle_VirtualBox_Extension_Pack-7.1.8.vbox-extpack
+VBoxManage list extpacks
+```
+Reboot your system after installation:
+
+```bash
+sudo reboot
+```
+
+## Deploy the BOSH Director with bosh-lite 
+Follow instruction in `Bosh-Lite` [website](https://bosh.io/docs/bosh-lite/)
+
+```bash
+git clone https://github.com/cloudfoundry/bosh-deployment ~/workspace/bosh-deployment
+mkdir -p ~/deployments/vbox
+cd ~/deployments/vbox
+```
+Create the BOSH Director VM:
+
+```bash
+bosh create-env ~/workspace/bosh-deployment/bosh.yml \
+--state ./state.json \
+-o ~/workspace/bosh-deployment/virtualbox/cpi.yml \
+-o ~/workspace/bosh-deployment/virtualbox/outbound-network.yml \
+-o ~/workspace/bosh-deployment/bosh-lite.yml \
+-o ~/workspace/bosh-deployment/bosh-lite-runc.yml \
+-o ~/workspace/bosh-deployment/uaa.yml \
+-o ~/workspace/bosh-deployment/credhub.yml \
+-o ~/workspace/bosh-deployment/jumpbox-user.yml \
+--vars-store ./creds.yml \
+-v director_name=bosh-lite \
+-v internal_ip=192.168.56.6 \
+-v internal_gw=192.168.56.1 \
+-v internal_cidr=192.168.56.0/24 \
+-v outbound_network_name=NatNetwork
+```
+Check that the VMs have been created in VirtualBox:
+
+```bash
+VBoxManage list vms
+```
+
+You should see entries similar to:
+
+```bash
+"sc-7ff79fac-045c-46e2-437f-ba09d54c40bd" {ca1d6ba0-e95d-4849-9d0f-f7723386951e}
+"vm-56718394-d4c1-4b94-6ce3-3d22c712844b" {56718394-d4c1-4b94-6ce3-3d22c712844b}
+```
+
+## Configure environment variables and login
+
+```bash
+export BOSH_CLIENT=admin
+export BOSH_CLIENT_SECRET=`bosh int ./creds.yml --path /admin_password`
+bosh alias-env vbox -e 192.168.56.6 --ca-cert <(bosh int ./creds.yml --path /director_ssl/ca)
+```
+
+Expected output:
+
+```bash
+Using environment '192.168.56.6' as client 'admin'
+
+Name               bosh-lite  
+UUID               d0207431-94e8-416e-a8ef-73f007d171c4  
+Version            282.0.4 (00000000)  
+Director Stemcell  -/1.822  
+CPI                warden_cpi  
+Features           config_server: enabled  
+                   local_dns: enabled  
+                   snapshots: disabled  
+User               admin  
+
+Succeeded
+helios02 :: 
+```
+## Set up routing for VM access
+Optionally, set up a local route for bosh ssh commands or accessing VMs
+directly:
+
+`sudo ip route add 10.244.0.0/16 via 192.168.56.6 # Linux (using iproute2 suite)`
```

Comment:
where do you get the subnet for `10.244.0.0/16`?

Reply:
From the official guide, it's the default address assigned to the deployed VM by bosh lite